addons/l10n_in/models/account_invoice.py

```diff
@@ -74,6 +75,20 @@ def _onchange_name_warning(self):
             }}
         return super()._onchange_name_warning()
 
+    @api.depends('invoice_line_ids.product_id', 'company_id.l10n_in_hsn_code_digit')
+    def _compute_hsn_code_warning(self):
+        for move in self:
+            product_lines = move.invoice_line_ids.filtered(lambda l: l.display_type == 'product')
+            if any(l.tax_ids for l in product_lines) and not move.company_id.vat:
+                move.l10n_in_hsn_code_warning = "Ensure that the GSTN number is included in the company information for the GST tax invoice."
```

Comment:
```suggestion
                move.l10n_in_hsn_code_warning = _("Ensure that the GSTN number is included in the company information for the GST tax invoice.")
``` 
Should we really complain about this here?  It is also calculating on non-Indian vendor bills/misc/...  It does not seem like we should complain about it here when it is missing.